github_metrics/app.py

```
@@ -1963,6 +1964,80 @@ async def get_metrics_trends(
+        raise HTTPException(
+            status_code=http_status.HTTP_400_BAD_REQUEST,
+            detail="PR number must be positive",
+        )
+
+    try:
+        story = await get_pr_story(db_manager, repository, pr_number)
+        if story is None:
+            raise HTTPException(
+                status_code=http_status.HTTP_404_NOT_FOUND,
+                detail=f"PR #{pr_number} not found in {repository}",
+            )
```

Ruff (0.14.6)

Abstract `raise` to an inner function

(TRY301)